shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -436,8 +437,20 @@ - (nullable id)accessibilityElementAtIndex:(NSInteger)index {
     return _semanticsObject;
   }
   SemanticsObject* child = [_semanticsObject children][index - 1];
-  if ([child hasChildren])
-    return [child accessibilityContainer];
+    
+    // This 'if' block handles adding accessibility support for the embedded platform view.
+    // We first check if the child is a semantic node for a platform view.
+    // If so, we add the platform view as accessibilityElements of the child.
+    shell::FlutterPlatformViewsController *flutterPlatformViewsController = _bridge.get()->flutter_platform_views_controller();
+    if (child.node.platformViewId > -1 && flutterPlatformViewsController) {
```

Comment:
What if `child` at a later point no longer has a platfromViewId? How do you remove the native a11y elements?

Reply:
We can clean it up when it happens, but I guess it might require some ugly tracking logic or some ugly type check. Is it even possible that the child object's platformViewId got changed but the child still exists in the tree?

Reply:
Added the code to clean up the a11y elements.